### PR TITLE
Calculator

### PR DESCRIPTION
**Pull request by AmitNahmias**
Last version I hope

### CODE REVIEW

File: Calculator.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                elif sign in dictionary.keys():
+                    result = dictionary[sign](int(self.list_of_exercise[self.list_of_exercise.index(sign) - 1]),
+                                              int(self.list_of_exercise[self.list_of_exercise.index(sign) + 1]))
+                    self.change_list(result, self.list_of_exercise.index(sign))
+                    i -= 2
+                i += 1
+        return self.list_of_exercise[0]
+
+
+def main():
```
Comment (guysl10) [inline Calculator.py]: It's common to write this docstream at the main:

```
"""
main() -> NoneType
Control the flow of the program
"""
```

File: Calculator.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
+class Calculator(object):
+    """ The class defines calculator with operations order according to specific instructions """
+
+    def __init__(self, list_of_exercise: list):
+        """
```
Comment (guysl10) [inline Calculator.py:5]: you didn't use EpyDoc format..

File: Calculator.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                                   self.dict_of_operations_prioritize_1]
+
+    def change_list(self, result, index: int):
+        """The function changing list values after calculate"""
+        self.list_of_exercise[index] = result
+        self.list_of_exercise.pop(index + 1)
+        self.list_of_exercise.pop(index - 1)
+
+    def exercise_handler(self):
+        """ This function handel in use of [ ] in the exercise and using calculate method"""
```
Comment (guysl10) [inline Calculator.py]: * **handle** and not handel

File: Calculator.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            temp_list.pop(0)
+            temp_list.pop(len(temp_list) - 1)
+            temp_instance_of_calc = Calculator(temp_list)
+            result = temp_instance_of_calc.calculate()
+            self.list_of_exercise.insert(opening, result)
+        else:
+            self.calculate()
+
+    def calculate(self):
+        """The function calculating exercise according specific math order"""
```
Comment (guysl10) [inline Calculator.py:54]: You don't have to start every sentence with " The function..." just write what's going on.
```
"""
Calculating exercise according to specific math order.
"""
```

File: Calculator.py (hunk 1 above)
Comment (guysl10) [inline Calculator.py]: The main is pretty massive.
The main must be between 5-7 lines, If it's more than this you should split it to another function.

**Review by guysl10 — CHANGES_REQUESTED** [review]
better than ever but still need to improve the doc handling.
You must use EpyDoc!

File: Calculator.py (hunk 3 above)
Comment (AmitNahmias) [reply Calculator.py]: Probably because the keyboard..

File: Calculator.py (hunk 4 above)
Comment (AmitNahmias) [reply Calculator.py:54]: Ok

File: Calculator.py (hunk 1 above)
Comment (AmitNahmias) [reply Calculator.py]: I will do it in my next program

**Comment by AmitNahmias** [comment]
Need improve

**Comment by AmitNahmias** [comment]
need improve

**Pull request closed without merging**